extensions/amp-live-list/0.1/live-list-manager.js

```diff
@@ -279,6 +293,17 @@ export class LiveListManager {
     // Polling may not be started yet if no live lists were registered by
     // doc ready in LiveListManager's constructor.
     if (liveList.isEnabled() && this.poller_ && this.ampdoc.isVisible()) {
+      if (this.enrolledInAppendRandomExperiment_ === null) {
+        // Origin Trial for cache busting requests for `amp-live-list`.
+        installOriginExperimentsForDoc(this.ampdoc);
+        originExperimentsForDoc(liveList.element)
+          .getExperiments()
+          .then(
+            trials =>
+              (this.enrolledInAppendRandomExperiment_ =
+                trials && trials.includes('amp-live-list-random'))
+          );
+      }
```

Comment:
:) This might need to be moved back (outside of the outer conditional) to address the case where `ampdoc.isVisible() == false` at the time `register()` is invoked. I think this could happen if a user switches to another tab immediately after opening an AMP page.

Yea this class's control flow is not the cleanest.